retrieve_assertion.go

```diff
@@ -66,17 +71,28 @@ func (sp *SAMLServiceProvider) RetrieveAssertionInfo(encodedResponse string) (*A
 		return nil, ErrMissingElement{Tag: AttributeStatementTag}
 	}
 
-	info := make(map[string]string)
-	for _, child := range attributeStatement.ChildElements() {
-		nameAttr := child.SelectAttr(NameAttr)
-		attributeValue := child.FindElement(childPath(child.Space, AttributeValueTag))
-		if attributeValue == nil {
-			return nil, ErrMissingElement{Tag: AttributeValueTag}
-		}
-		info[nameAttr.Value] = attributeValue.Text()
+	doc := etree.NewDocument()
+	doc.SetRoot(attributeStatement)
+	bs, err := doc.WriteToBytes()
+
+	if err != nil {
+		return nil, err
+	}
+
+	var attrs struct {
+		List []Attribute `xml:"Attribute"`
+	}
+	err = xml.Unmarshal(bs, &attrs)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I'm curious your motivation for switching from etree to encoding/xml here - just simplicity of the unmarshal?

Relatedly, do we ned to handle a situation where an `Attribute` has an unexpected namespace prefix?